ESP-IDF-Subordinate/main/file.c

```diff
@@ -0,0 +1,133 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+#include "freertos/semphr.h"
+#include "esp_now.h"
+
+#include "file.h"
+
+
+static struct{
+
+    uint8_t dev_mac_address[6];
+    bool disable_sleep;
+    uint32_t tx_nack_timeout_ms;
+    data_to_transmit _data_to_transmit;
+    on_data_rx _on_data_rx;
+    sub_transmit_periods_t *transmit_periods;
+    QueueHandle_t rx_data;
+    SemaphoreHandle_t sleep_lock;
+    uint8_t *tx_buffer;
+    uint8_t tx_data_size;
+    esp_now_send_status_t last_message_status;
+    uint16_t interval_ms;
+    SemaphoreHandle_t mutex;
+
+} sub_struct;
+
+sub_status_t init(sub_init_struct_t *init_struct, sub_transmit_periods_t *transmit_periods){
+    
+    if(init_struct == NULL || transmit_periods == NULL){
+        return SUB_NULL_ERR;
+    }
+
+    sub_struct.disable_sleep = init_struct->disable_sleep;
+
+    if(init_struct->tx_nack_timeout_ms > 60000){
```

Comment:
Tutaj raczej chcemy ustawić ponad 60 sekund, bo cały lot wraz z opadaniem na spadochronie trwa kilka minut.

Reply:
A teraz myślę że w sumie możemy zahardkodować ten czas. Natomiast checki nadal bym zostawił i zapisywał to w tej strukturze. Kwestia tylko że będzie przypisywana z jakiegoś #define a nie ze struktury konfiguracyjnej.